libusb/hid.c

```diff
@@ -546,8 +558,10 @@ static void fill_device_info_usage(struct hid_device_info *cur_dev, libusb_devic
 		get_usage(hid_report_descriptor, res,  &page, &usage);
 	}
 
-	cur_dev->usage_page = page;
-	cur_dev->usage = usage;
+	if (cur_dev) {
```

Comment:
early return here as well - it doesn't make sense to call this function with NULL cur_dev at all